lasagne/tests/layers/test_base.py

```diff
@@ -145,3 +147,67 @@ def test_get_output_shape_for_notimplemented(self, layer):
     def test_get_output_for_notimplemented(self, layer):
         with pytest.raises(NotImplementedError):
             layer.get_output_for(Mock())
+
+
+class TestUnLayer:
+
+    @pytest.fixture
+    def unlayer_vars(self):
+        from lasagne.layers.dense import DenseLayer
+        from lasagne.layers.input import InputLayer
+        from lasagne.layers.base import UnLayer
+        from lasagne.nonlinearities import identity
+
+        b = None
+        nonlinearity = identity
+
+        shape = (10, 12)
+        l_in = InputLayer(shape=shape)
+
+        layer = DenseLayer(
+            l_in,
+            num_units=3,
+            b=b,
+            nonlinearity=nonlinearity,
```

Comment:
I'd directly write `b=None` and `nonlinearity=identity` here, as that's very essential for your `test_get_output_for()`.